### PR TITLE
WIP: Verilator

### PR DESCRIPTION
**Pull request by drichmond**
This PR adds support for Verilator simulation. 

It depends on PRs:

[274](https://github.com/bespoke-silicon-group/basejump_stl/pull/274) in BaseJump
[279](https://github.com/bespoke-silicon-group/basejump_stl/pull/279) in BaseJump

(These PRs are combined in the [manycore_verilator](https://github.com/bespoke-silicon-group/basejump_stl/tree/manycore_verilator) branch)

[295](https://github.com/bespoke-silicon-group/bsg_manycore/pull/295) in Manycore
[307](https://github.com/bespoke-silicon-group/bsg_manycore/pull/307) in Manycore

(These PRs are combined in the [verilator](https://github.com/bespoke-silicon-group/bsg_manycore/tree/verilator) branch)

At the moment, this PR does not support generating waveforms (verilator generates VCD files, which is inefficient), or profiler outputs. 

The profilers are not supported because `forever \n @posedge clk` statements are not supported in Verilator. When that is fixed, these can be re-added.

You can run verilator simulation using the [verilator](https://github.com/bespoke-silicon-group/bsg_bladerunner/tree/verilator) branch in Bladerunner

This PR passes regression on 4x4_fast_n_fake

(EDIT: This PR passes, but it does NOT support multiple init/cleanup runs -- so two regression tests have to be cut out)

### CODE REVIEW

**Comment by drichmond** [comment]
I don't expect this PR to be merged. It breaks everything for VCS-based simulation and FPGA compilation. There's a fair amount of work to get this into mainline (if that's what we'd like to do)

**Comment by leonardxiang** [comment]
WoW, It Works~Thanks Dustin!

**Comment by drichmond** [comment]
@leonardxiang You got it to work!?

**Comment by leonardxiang** [comment]
@drichmond Yes, on my centos matchine. It is very helpful!

File: libraries/bsg_manycore_machine.cpp [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+/**
+ * Receive a packet from manycore hardware
+ * @param[in] mc       A manycore instance initialized with hb_mc_manycore_init()
+ * @param[in] response A packet into which data should be read
+ * @param[in] timeout  A timeout counter. Unused - set to -1 to wait forever.
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_machine_receive(hb_mc_manycore_t *mc,
+                          hb_mc_packet_t *packet,
```
Comment (drichmond) [inline libraries/bsg_manycore_machine.cpp:86]: This needs to be fixed

**Pull request closed without merging**